controllers/btpoperator_controller.go

```diff
@@ -446,33 +449,122 @@ func (r *BtpOperatorReconciler) HandleProcessingState(ctx context.Context, cr *v
+}
+
+func (r *BtpOperatorReconciler) createUnstructuredObjectsFromManifestsDir(ctx context.Context, manifestsDir string) ([]*unstructured.Unstructured, error) {
+	logger := log.FromContext(ctx)
+
+	objs, err := r.manifestHandler.CollectObjectsFromDir(manifestsDir)
+	if err != nil {
+		logger.Error(err, "while collecting objects from manifests")
+		return nil, err
+	}
+	us, err := r.manifestHandler.ObjectsToUnstructured(objs)
+	if err != nil {
+		logger.Error(err, "while converting objects to Unstructured")
+		return nil, err
+	}
+
+	return us, nil
+}
+
+func (r *BtpOperatorReconciler) prepareModuleResources(ctx context.Context, us []*unstructured.Unstructured) *ErrorWithReason {
+	logger := log.FromContext(ctx)
+	logger.Info("preparing module resources")
+
+	var configMapIndex, secretIndex int
+	for i, u := range us {
+		if u.GetName() == btpServiceOperatorConfigMap && u.GetKind() == configMapKind {
+			configMapIndex = i
+		}
+		if u.GetName() == btpServiceOperatorSecret && u.GetKind() == secretKind {
+			secretIndex = i
 		}
-		if ready {
-			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateReady, ReconcileSucceeded, "Reconcile succeeded")
+	}
+
+	secret, errorWithReason := r.getAndVerifyRequiredSecret(ctx)
+	if errorWithReason != nil {
+		return errorWithReason
+	}
+
+	r.addLabels(us...)
+	r.setNamespace(us...)
+	r.deleteCreationTimestamp(us...)
+	if err := r.setConfigMapValues(secret, us[configMapIndex]); err != nil {
+		logger.Error(err, "while setting ConfigMap values")
+		return NewErrorWithReason(PreparingModuleResourcesFailed, "Failed to set ConfigMap values")
+	}
+	if err := r.setSecretValues(secret, us[secretIndex]); err != nil {
+		logger.Error(err, "while setting Secret values")
+		return NewErrorWithReason(PreparingModuleResourcesFailed, "Failed to set Secret values")
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, us []*unstructured.Unstructured) error {
+	logger := log.FromContext(ctx)
+	logger.Info("reconciling module resources")
+
+	for _, u := range us {
+		if err := r.Patch(ctx, u, client.Apply, client.ForceOwnership, client.FieldOwner(operatorName)); err != nil {
+			return fmt.Errorf("while reconciling %s %s: %w", u.GetName(), u.GetKind(), err)
 		}
 	}
 
 	return nil
 }
 
+func (r *BtpOperatorReconciler) waitForResourcesReadiness(ctx context.Context, us []*unstructured.Unstructured) error {
+	numOfResources := len(us)
+	resourcesReadinessInformer := make(chan<- bool, numOfResources)
+	allReadyInformer := make(chan bool)
+	for _, u := range us {
+		go r.checkResourceReadiness(ctx, u, resourcesReadinessInformer)
+	}
+	go func(c chan<- bool) {
+		now := time.Now()
+		for {
+			if time.Since(now) >= ReadyTimeout {
+				return
+			}
+			if len(resourcesReadinessInformer) == numOfResources {
+				allReadyInformer <- true
+				return
+			}
+			time.Sleep(ReadyCheckInterval)
+		}
+	}(resourcesReadinessInformer)
```

Comment:
nit: I think it should be possible to implement this without polling. If I follow this correctly, as implemented, there is a single channel `resourcesReadinessInformer` with the capacity equal to `number of resources`. It starts goroutine per resource which, when a particular resource is ready, sends message to channel. Here it polls the number of messages unread in the channel.

Consider something along the lines of receiving `number of resources` messages and then forwarding immediately to `allReadyInformer`. That way you save on some polling.

I would suggest something along the lines of:
```suggestion
    go func(c chan<- bool) {
        timeout := time.After(ReadyTimeout)
        // expect to receive N messages
        for i:=0; i < numOfResources; i++{
            select {
            case <-resourcesReadinessInformer:
              // some resource ready
            case <-timeout:
              // reached timeout
              return
            }
        }
        // received N messages, all are ready
        allReadyInformer <- true
    }
```

Reply:
love it! 🚀 let me apply your suggestion

Reply:
applied in https://github.com/kyma-project/btp-manager/pull/98/commits/2ff3c8bf5356b6f6d000a0ad6fe661cd88099621